### PR TITLE
[go_router] add `parseExtra` in `GoRouterState` 

### PR DESCRIPTION
**Pull request by marcusedu**
The method `parseExtra` implemented to handle Objects in `extra` field on `GoRouterState`.
`parseExtra` checks the `T` expected for the user and search in `extra`, analyzing if `extra` is Map or List and search inside for the `T` type and `key` and return the value.

```dart
context.go('/check_out', extra: [payment, items, account]);

GoRoute(
  path: '/check_out',
  builder: (context, state) => CheckOutScreen(
    payment: state.parseExtra<Payment>()!,
    items: state.parseExtra<List<Item>>()!,
    account: state.parseExtra<Account>()!,
  ),
);

// Or
context.go('/check_out', extra: {
  'payment': payment,
  'items': items,
  'account': account
});

GoRoute(
  path: '/check_out',
  builder: (context, state) => CheckOutScreen(
    payment: state.parseExtra<Payment>('payment')!,
    items: state.parseExtra<List<Item>>('items')!,
    account: state.parseExtra<Account>('account')!,
  ),
);
```

### CODE REVIEW

File: packages/go_router/lib/src/go_router_state.dart [hunk 1]
```diff
@@ -56,6 +56,59 @@ class GoRouterState {
   /// An extra object to pass along with the navigation.
   final Object? extra;
 
+  /// Handle extra object passed along with the navigation.
```
Comment (nilsreichardt) [inline packages/go_router/lib/src/go_router_state.dart:59]:
```suggestion
  /// Handle extra object passed along with the navigation.
  ///
```

https://dart.dev/guides/language/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph

File: packages/go_router/lib/src/go_router_state.dart [hunk 2]
```diff
@@ -56,6 +56,59 @@ class GoRouterState {
...
+  ///  'account': Account(
+  ///     id: '1',
+  ///     name: 'Jack Smith',
+  ///     email: 'jack.smith@email.com'),
+  ///  'someOtherObject': someOtherObject,
+  ///  };
+  ///
+  /// context.push(HomeScreen.route, extra: extra);
+  ///
+  ///final goRoute = GoRoute(
```
Comment (nilsreichardt) [inline packages/go_router/lib/src/go_router_state.dart:78]:
```suggestion
  /// final goRoute = GoRoute(
```

**Comment by marcusedu** [comment]
Hey @nilsreichardt please accept the cla/google in this link to solve one remain issue.
https://cla.developers.google.com/

**Comment by nilsreichardt** [comment]
Weird, I already accepted the CLA:
image

I also contributed already to this repository: https://github.com/flutter/packages/pulls?q=is%3Apr+sort%3Aupdated-desc+author%3Anilsreichardt+is%3Aclosed

**Comment by nilsreichardt** [comment]
@marcusedu I think I found the issue. Can re-trigger the CI? You could commit an empty commit with `git commit --allow-empty -m "Re-trigger CI"`

**Comment by chunhtai** [comment]
I think this pr makes a relatively simple task to be more descriptive, which is a good thing, but we opens up an API that we will have to maintain for the future. This API also relies on the fact the extra needs to be a map. I think PR may not worth in terms of maintenance cost. I will close this pr, but feel free to reopen it if you think otherwise

**Pull request closed without merging**